debian/couchdb.postrm

```diff
@@ -19,19 +19,30 @@ case $1 in
         if test -d "/opt/couchdb/etc"; then
             rm -rf "/opt/couchdb/etc"
         fi
-        if test -d "/var/lib/couchdb"; then
-            rm -rf "/var/lib/couchdb"
-        fi
         if test -d "/var/log/couchdb"; then
             rm -rf "/var/log/couchdb"
         fi
-        pkill -u couchdb >/dev/null 2>&1 || true
-        if getent passwd couchdb > /dev/null; then
-            deluser couchdb >/dev/null 2>&1
-        fi
-        if getent group couchdb > /dev/null; then
-            delgroup couchdb >/dev/null 2>&1
+
+        db_input high couchdb/postrm_remove_databases || true
+        db_go || true
+        db_get couchdb/postrm_remove_databases || true
+
+        if [ "$RET" = "true" ]; then
+            if test -d "/var/lib/couchdb"; then
+                rm -rf "/var/lib/couchdb"
+            fi
+
+            # If we didn't delete the data, we should
+            # keep the uid and gid too.
+            if getent passwd couchdb > /dev/null; then
+                deluser couchdb >/dev/null 2>&1
+            fi
+            if getent group couchdb > /dev/null; then
+                delgroup couchdb >/dev/null 2>&1
+            fi
         fi
+
+        pkill -u couchdb >/dev/null 2>&1 || true
         db_reset couchdb/mode
         db_reset couchdb/nodename
         db_reset couchdb/cookie
```

Comment:
probably unrelated entirely but why do we forcibly reset the debconf settings after purge? I think we should not reset the `couchdb/nodename` parameter if we're not purging the databases, but I don't know about the rest.

Reply:
This sequence of events should be possible, and not result in a surprise:

```bash
apt install couchdb
# configure as single node with an admin user for testing, password "admin"
# do some testing, create some DBs
apt purge couch
apt install couchdb
# re-configure couchdb for a cluster with a real admin user/password combo
```

Have you tried this? I should test it myself before I +1 this.